app/services/saml_endpoint.rb

```diff
@@ -1,11 +1,41 @@
 # frozen_string_literal: true
 
 class SamlEndpoint
-  SAML_YEARS = AppArtifacts.store.members.map(&:to_s).map do |key|
-    regex = /saml_(?<year>\d{4})_(?<key_cert>key|cert)/
-    matches = regex.match(key)
-    matches && matches[:year]
-  end.compact.uniq.freeze
+  SAML_YEARS = IdentityConfig.store.saml_endpoint_configs.map do |config|
+    config.fetch(:suffix).to_s
+  end.uniq.sort.freeze
+
+  SAML_YEAR_CERTS = SAML_YEARS.each_with_object({}) do |year, map|
+    x509_cert =
```

Comment:
[question] there's quite a lot of logic in these constants. i'm curious if we should make them methods (or even a class, although that might be overkill) rather than trying to do all this work here?

Reply:
Yeah, I was trying to figure out the best way to do that. To use methods, they have to be defined before the constants, and another class does feel like too much.

Reply:
I think the best two options are move the constants to the bottom of the file and define method for the setup (as we do [here](https://github.com/18F/identity-idp/blob/faa62c5c935b2059ad6353afeda78331890a95b9/app/services/rate_limiter.rb#L289)) or have long constants like it is now. Perhaps the first option is preferable?

Reply:
hm, yeah! i think that would be nicer. maybe easier to test directly too?

Reply:
That's a really good point, will update.